src/G2/Language/Naming.hs

```diff
@@ -179,7 +179,7 @@ altMatchNames (DataAlt dc i) = dataConName dc ++ (map idName i)
 altMatchNames _ = []
 
 dataConName :: DataCon -> [Name]
-dataConName (DataCon n _) = [n]
+dataConName (DataCon n _ u e) = [n] ++ map idName u ++ map idName e
```

Comment:
We just want to return the data constructor name here, we do not want to return the type names (revert this change)

Reply:
That is, just want to return `n` (notice how this function is used- the only functions that call it are just supposed to return value level names, not type level names.)

Reply:
I got it, sorry for my mistakes. I think because I was updating this along with the containedASTs which make me think we should get all the names from the function. I assume those three changes and changes I need to make before merging into the master right?